lib/index.js

```diff
@@ -87,31 +75,67 @@ class Subscriber extends events.EventEmitter {
     // Here we should get all the subscriptions under the channel topic and unsubscribe them
   }
 
-  _createSubscription(topic,channel) {
+  _createSubscription(topic, channel) {
 
     const subscriptionUUID = uuid.v1();
     var subscriptionName = `${namePrefix}-${channel}-${subscriptionUUID}`;
 
     topic.subscribe(subscriptionName, (err, subscription) => {
+
       if (err) {
         console.log(`Failed to create subscription ${err}`);
         return;
       }
 
       console.log(`Subscription ${subscription.name} created.`);
-      process.on('SIGTERM', () => {
-        subscription.delete();
-      });
-      process.on('SIGINT', () => {
-        subscription.delete();
-      });
-      subscription.on('message', (message) => {
-        if (message.ackId) {
-          message.ack();
+      
+      function deleteSubscription() {
+        removeListeners();
+        console.log('Subscriber: SIGTERM received, deleting subscription');
```

Comment:
Nits, but SIGTERM is misleading here since it's also called on SIGINT. You could just say "signal received"